src/store/membership_fee/actions.ts

```diff
@@ -38,13 +38,19 @@ const fieldToValidationMutation = new Map( [
 	[ 'type', SET_TYPE_VALIDITY ],
 ] );
 
-const validateFeeOnClientSide = ( fee: string ): Validity => Helper.isNonNumeric( fee ) ? Validity.INVALID : Validity.VALID;
+const validateFeeOnClientSide = ( context, fee: string ): Validity => {
```

Comment:
This helper function now is no longer pure and has a side effect but its name still makes it look like a helper function (only passing in the context is an indicator that it does more).
Since the calling code commits the fee anyway, I'd rather leave this helper as-is and change the calling code (example):

```typescript
const feeValidity = validateFeeOnClientSide( initialData.fee );
context.commit( SET_FEE_VALIDITY, feeValidity );
context.commit( SET_FEE, feeValidity === Validity.VALID ? initalData.fee : '' );
``` 
That way we also avoid changing the store twice